pom.xml

```diff
@@ -102,6 +103,22 @@
             </configuration>
           </execution>
         </executions>
+        <configuration>
+          <rules>
+            <requireUpperBoundDeps>
+              <excludes combine.children="append">
+                <!-- promoted-builds requests 1.9.2, core requests 1.8.4 -->
+                <exclude>org.apache.ant:ant</exclude>
```

Comment:
Thanks for the pointer.  I've updated the pull request with a few more changes trying to do what I believe you were suggesting.  Let me know if I've misunderstood.

I excluded the transitive dependency on objenesis from the mockito dependency, having it rely on the version that will be included by powermock (since powermock provides the newer version).  That seems very low risk, since that is purely a test dependency.

I excluded the transitive dependency on org.apache.ant from the optional promoted-builds dependency because I assume that if promoted-builds is installed in the system, then it will deliver the best set of dependencies for its needs.  If promoted-builds is not installed, then the org.apache.ant dependency from core is the better choice.